src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -69,7 +69,18 @@ internal IEnumerable<MessageMetadata> GetAllMessages()
             return new List<MessageMetadata>(messages.Values);
         }
 
-        internal MessageMetadata RegisterMessageType(Type messageType)
+        internal void RegisterAllMessageTypes(IList<Type> availableTypes)
+        {
+            foreach (var type in availableTypes)
+            {
+                if (conventions.IsMessageType(type))
```

Comment:
I like the idea of moving the conventions check here, but I'm wondering if there might be a better name for the method. To me, `RegisterAllMessageTypes` doesn't convey that it's going to take the types you've passed to it, and selectively add them if they pass the conventions check. The original `RegisterMessageType` method would register any type you passed to it, so it seems like making this clearer would be a good idea.